src/acp/policy/getPolicyUrlAll.test.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { ACP } from "../constants";
+import {
+  DEFAULT_ACCESS_CONTROL_RESOURCE_URL,
+  TEST_URL,
+} from "../mock/constants";
+import { mockAccessControlledResource } from "../mock/mockAccessControlledResource";
+import { createDatasetFromSubjects } from "../mock/dataset";
+import { getPolicyUrlAll } from "./getPolicyUrlAll";
+
+describe("getPolicyUrlAll()", () => {
+  it("Returns empty array for empty Access Control Resource", async () => {
+    const resource = mockAccessControlledResource();
+
+    expect(getPolicyUrlAll(resource)).toStrictEqual([]);
+  });
+
+  it("Returns a member policy URL when present", async () => {
+    const resource = mockAccessControlledResource(
+      createDatasetFromSubjects([
+        [
+          DEFAULT_ACCESS_CONTROL_RESOURCE_URL,
+          [[ACP.accessControl, [TEST_URL.accessControl1]]],
+        ],
+        [
+          TEST_URL.accessControl1,
+          [[ACP.apply, [TEST_URL.accessControl1Policy1]]],
+        ],
+      ])
+    );
+
+    expect(getPolicyUrlAll(resource)).toStrictEqual([
+      TEST_URL.accessControl1Policy1,
+    ]);
+  });
+
+  it("Returns all member policy URLs when present", async () => {
```

Comment:
Shouldn't this test for both policies and access policies ?

Reply:
```suggestion
  it("Returns all policy URLs when present", async () => {
```

I know this is confusing.
The convention in use (and previously in use) is that this function returns only the policies that apply to the resource, not the ones applying to its ACR.

Reply:
Okay, then maybe a comment clarifying that this is intended would help :)